### PR TITLE
Added recipe for Genio

### PR DESCRIPTION
**Pull request by jackburton79**
Added recipe for Genio

### CODE REVIEW

File: haiku-apps/genio/genio-1.0.0~beta.recipe [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+SOURCE_URI="https://github.com/Genio-The-Haiku-IDE/Genio/archive/v1.tar.gz"
+CHECKSUM_SHA256="7ae9a8affd59869741b7a891384fc733ce78fd404ba2e433708a68a473d11358"
+SOURCE_FILENAME="Genio-v1.tar.gz"
+SOURCE_DIR="Genio-1"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	Genio${secondaryArchSuffix} = $portVersion
```
Comment (Begasus) [inline haiku-apps/genio/genio-1.0.0~beta.recipe:20]: Should reflect recipe's name (no capitols):
```diff
-	Genio${secondaryArchSuffix} = $portVersion
+	genio${secondaryArchSuffix} = $portVersion
```

Comment (jackburton79) [reply haiku-apps/genio/genio-1.0.0~beta.recipe:20]: Also in the line "app:Genio..." ?

Comment (Begasus) [reply haiku-apps/genio/genio-1.0.0~beta.recipe:20]: No, "app:Genio" is fine :)

File: haiku-apps/genio/genio-1.0.0~beta.recipe [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+BUILD_PREREQUIRES="
+	makefile_engine
+	cmd:gcc${secondaryArchSuffix}
+	cmd:make
+	"
+
+BUILD()
+{
+	make OBJ_DIR=objects \
+		BUILDHOME=`finddir B_SYSTEM_DEVELOP_DIRECTORY`
```
[the remark this answers is not in the document]
Comment (Begasus) [reply haiku-apps/genio/genio-1.0.0~beta.recipe:39]: This could be reduced to 1 line (wrap lines is now up to 100 characters)

**Comment by Begasus** [comment]
There is now a `data/templates` directory in `$appsDir`, that shouldn't be there, preferably installed to `$dataDir` if possible.

**Comment by jackburton79** [comment]
> Can't wait to try this one, thanks! (@jackburton79, linter complains of trailing spaces in the recipe. There are 3 on `INSTALL()`, lines 47, 49, and 52).



> There is now a `data/templates` directory in `$appsDir`, that shouldn't be there, preferably installed to `$dataDir` if possible.

At the moment it isn't possible, since Genio looks in data/templates.

File: haiku-apps/genio/genio-1.0.0~beta.recipe (hunk 1 above)
Comment (jackburton79) [reply haiku-apps/genio/genio-1.0.0~beta.recipe:20]: > No, "app:Genio" is fine :)
> 
> Err, without secondaryArchSuffix

Weird: I copied from scintilla and there they use 
```
lib:libscintilla$secondaryArchSuffix = $portVersion
```

**Comment by jackburton79** [comment]
> > Can't wait to try this one, thanks! (@jackburton79, linter complains of trailing spaces in the recipe. There are 3 on `INSTALL()`, lines 47, 49, and 52).
> 
> > There is now a `data/templates` directory in `$appsDir`, that shouldn't be there, preferably installed to `$dataDir` if possible.
> 
> At the moment it isn't possible, since Genio looks in a data/templates subdirectory of the application directory. I guess we could change that in a next release

I guess we could do this:
instead of

```
install -m 0755 -d "$appsDir"
install -m 0755 -t "$appsDir" app/Genio
install -m 0755 -d "$appsDir/data"
```

we could create a subdirectory "Genio" where we put all needed stuff (like Paladin does)
```

install -m 0755 -d "$appsDir"
install -m 0755 -d "$appsDir/Genio"
install -m 0755 -t "$appsDir/Genio" app/Genio
install -m 0755 -d "$appsDir/Genio/data"

cp -a data/templates "$appsDir/Genio/data"
```

Would that be acceptable ?

**Comment by Begasus** [comment]
> Would that be acceptable ?

That would be my next suggestion if it needs data to be present near the binary.

File: haiku-apps/genio/genio-1.0.0~beta.recipe (hunk 1 above)
Comment (Begasus) [reply haiku-apps/genio/genio-1.0.0~beta.recipe:20]: lib:libfoo$secondaryArchSuffix are libraries installed into $prefix/lib/x86, those installed in $prefix/lib don't need the suffix.
For binaries placed in $appsDir this is the same, no suffix is needed here.

**Comment by Begasus** [comment]
Builds and runs fine on 32bit (checked a build in there from one of the templates).
Only missing an option to switch compiler on 32bit, I guess it defaults to primary (gcc2 for 32bit)?

File: haiku-apps/genio/genio-1.0.0~beta.recipe (hunk 1 above)
Comment (jackburton79) [reply haiku-apps/genio/genio-1.0.0~beta.recipe]: Sorry, so to clean up things, is this correct ?
Also if it only works with the secondary arch x86, and not x86_gcc2 ?

```
genio$ = $portVersion
app:Genio = $portVersion
```

**Comment by Begasus** [comment]
> > Builds and runs fine on 32bit (checked a build in there from one of the templates). Only missing an option to switch compiler on 32bit, I guess it defaults to primary (gcc2 for 32bit)?
> 
> It only works on x86, not x86_gcc2. The recipe already takes that into account (copied from scintilla)

Looking at "gcc -v" when running the "Run console program" shows me gcc2?

Genio

**Comment by jackburton79** [comment]
> > > Builds and runs fine on 32bit (checked a build in there from one of the templates). Only missing an option to switch compiler on 32bit, I guess it defaults to primary (gcc2 for 32bit)?
> > 
> > 
> > It only works on x86, not x86_gcc2. The recipe already takes that into account (copied from scintilla)
> 
> Looking at "gcc -v" when running the "Run console program" shows me gcc2?
> 
> Genio

Of course: you can use Genio to develop FOR x86_gcc2, but it only builds WITH newer gcc.

File: haiku-apps/genio/genio-1.0.0~beta.recipe (hunk 1 above)
Comment (Begasus) [reply haiku-apps/genio/genio-1.0.0~beta.recipe]: We are building for secondary architecture (primary marked broken !x86_gcc2) so:
```
genio$secondaryArchSuffix
app:Genio = $portVersion
```

**Comment by Begasus** [comment]
> Of course: you can use Genio to develop FOR x86_gcc2, but it only builds WITH newer gcc.

It would be nice to see it able to build for `x86 (gcc11)`, in Terminal we can do this with `setarch x86`.

File: haiku-apps/genio/genio-1.0.0~beta.recipe (hunk 1 above)
Comment (jackburton79) [reply haiku-apps/genio/genio-1.0.0~beta.recipe]: Done.

**Comment by Begasus** [comment]
LGTM 👍

**Comment by Begasus** [comment]
> Usually other apps are capitalized.

I don't know how that is done in the depot, but looking at the one for TuxPaint I also see it as lowercase.

**Pull request merged**